cli/src/pcluster/validators/networking_validators.py

```diff
@@ -57,6 +64,58 @@ def _validate(self, subnet_ids: List[str]):
             self._add_failure(str(e), FailureLevel.ERROR)
 
 
+class QueueSubnetsValidator(Validator):
+    """
+    Queue Subnets validator.
+
+    Check that there is no duplicate subnet id in the subnet_ids list.
+    Check that subnets in a queue belong to different AZs (EC2 Fleet requests do not support multiple subnets
+    in the same AZ).
+    """
+
+    def _validate(self, queue_name, subnet_ids: List[str], subnet_id_az_mapping: dict):
+
+        # Test if there are duplicate IDs in subnet_ids
+        if len(set(subnet_ids)) < len(subnet_ids):
+            duplicate_ids = [key for key, value in Counter(subnet_ids).items() if value > 1]
+            self._add_failure(
+                "The following subnet ids are specified multiple times "
+                "in queue {0}: {1}.".format(
+                    queue_name,
+                    ", ".join(duplicate_ids),
+                ),
+                FailureLevel.ERROR,
+            )
+
+        # Test if the subnets are all in different AZs
+        else:
+            try:
+                az_set = {subnet_id_az_mapping[subnet_id] for subnet_id in subnet_ids}
+                if len(az_set) < len(subnet_ids):
+
+                    # Find the AZs with multiple subnets
+                    azs_with_multiple_subnets = {}
```

Comment:
Could we move this logic into a separate function, named `get_subnets_sharing_an_az` perhaps.

Also, could it be simplified to avoid the nested loop?
```
def get_subnets_sharing_an_az(subnet_ids):
    subnets_sharing_az = set()
    visited_subnet_azs = {}

    for subnet_id in subnet_ids:
        az = subnet_id_az_mapping[subnet_id]

        if visited_subnet_azs.get(subnet_id) == az:
            subnets_sharing_az.add(subnet_id)
        else:
            visited_subnet_azs[subnet_id] = az

    return list(subnets_sharing_az)
```

Reply:
We discussed about this offline and decided to implement a new property `az_subnet_ids_mapping` in the `_QueueNetworking` class. This property is a dictionary containing the list of subnets belonging to each AZ used in the queue.